packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -180,33 +194,48 @@ class RenderParagraph extends RenderBox {
     // us to validate our intrinsic sizes, which will change _textPainter's
     // layout because the intrinsic size calculations are destructive.
     final Size textSize = _textPainter.size;
-    size = constraints.constrain(textSize);
+    final double effectiveLines =
+      // (overflow == TextOverflow.ellipsis && maxLines == null) ? 1.0 :
+      (maxLines?.toDouble() ?? double.INFINITY);
+    final double constrainedHeight =
+      textSize.height.clamp(0.0, _textPainter.preferredLineHeight * effectiveLines);
```

Comment:
OK, went with @Hixie 's suggestion and added new getter: https://github.com/flutter/engine/pull/3342